skills/__tests__/hears.test.js

```diff
@@ -0,0 +1,108 @@
+const { helpHandler, extractMessageContents } = require('../hears');
+const { MAIN_HELP_TEXT } = require('../../static');
+
+jest.mock('../../lib/asyncBot');
+
+const TEST_USER_ID = 'TEST_USER';
+const TEST_USER_NAME = 'TEST_USER_NAME';
+const LEARNER_OBJECT = { id: TEST_USER_ID, name: TEST_USER_NAME };
+
+const TEACHER_1_ID = 'TEACHER1';
+const TEACHER_1_NAME = 'TEACHER_1_NAME';
+const TEACHER_2_ID = 'TEACHER2';
+const TEACHER_2_NAME = 'TEACHER_2_NAME';
+
+const mockBot = new (jest.fn())();
+mockBot.whisper = jest.fn();
+
+// These all get passed to asyncBot in extractMessageContents
+mockBot.api = jest.fn();
+mockBot.api.users = jest.fn();
+mockBot.api.users.info = jest.fn(({ user }) => {
+  let name;
+  if (user === TEST_USER_ID) { name = TEST_USER_NAME; }
+  if (user === TEACHER_1_ID) { name = TEACHER_1_NAME; }
+  if (user === TEACHER_2_ID) { name = TEACHER_2_NAME; }
+  return Promise.resolve({ user: { name } });
+});
+
+const getMessage = text => ({
+  event: { text },
+  user: TEST_USER_ID,
+});
+
+describe('Test the helpHandler', () => {
+  test('helpHandler whispers the given message', () => {
+    const messageObject = { messageInfo: 'stuff' };
+    helpHandler(mockBot, messageObject);
+    expect(mockBot.whisper.mock.calls.length).toBe(1);
+    expect(mockBot.whisper.mock.calls[0][0]).toBe(messageObject);
+    expect(mockBot.whisper.mock.calls[0][1]).toBe(MAIN_HELP_TEXT);
+  });
+});
+
+describe('Test extractMessageContent', () => {
+  test('no skill or teacher mentioned', () => {
+    const message = getMessage('Message with no teacher and no skill');
+    return expect(extractMessageContents(mockBot, message)).resolves.toMatchObject({
+      learnerObject: LEARNER_OBJECT,
+      teacherObjects: [],
+      skills: [],
+    });
+  });
+
+  test('one skill and no teacher mentioned', () => {
+    const message = getMessage('Message with no teacher and one _skill_');
+    return expect(extractMessageContents(mockBot, message)).resolves.toMatchObject({
+      learnerObject: LEARNER_OBJECT,
+      teacherObjects: [],
+      skills: ['skill'],
+    });
+  });
```

Comment:
This is where I personally would have started following Arrange, Act, Assert for closely.
```javascript
  test('one skill and no teacher mentioned', async () => {
    const message = getMessage('Message with no teacher and one _skill_');

    const result = await extractMessageContents(mockBot, message)

    expect(result).toMatchObject({
      learnerObject: LEARNER_OBJECT,
      teacherObjects: [],
      skills: ['skill'],
    });
  });

Reply:
Yep, totally on board with this, gonna change them up to set a good precedence.